src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { TodoList } from '../TodoList';
+import { getTodos } from '../../api/todos';
+import { useTodosContext } from '../store';
+import { useUncomplitedTodos } from '../../helpers/useUncomplitedTodos';
+import { TodoFilter } from '../TodoFilter';
+import { useComplitedTodos } from '../../helpers/useComplitedTodos';
+
+const USER_ID = 12027;
+
+export const TodoApp: React.FC = () => {
+  const { todos, reciveTodos } = useTodosContext();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState('');
+
+  const compTodosLength = useComplitedTodos().length;
+  const uncompTodosLength = useUncomplitedTodos().length;
+
+  const resetHasError = () => {
+    setHasError('');
+  };
+
+  const handlerErrors = useCallback(
+    (errMessage: string) => {
+      setHasError(errMessage);
+
+      setTimeout(() => resetHasError(), 3000);
+    },
+    [],
+  );
+
+  useEffect(() => {
+    resetHasError();
+
+    getTodos(USER_ID)
+      .then(todosFS => {
+        reciveTodos(todosFS);
+      })
+      .catch(() => handlerErrors('Unable to load todos'))
+      .finally(() => setIsLoading(false));
+  }, [handlerErrors, reciveTodos]);
+
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
```

Comment:
You can move it to Header component

Reply:
That is idea, less code in main component.